File: Solo Learn/Médio/Palavras Fluentes/flowing_words.java
import java.util.Scanner;

public class SentenceFlow {

    public static String checkSentenceFlow(String sentence) {
        String[] words = sentence.toLowerCase().split(" ");  // Converte para minúsculas e divide a frase em palavras

        for (int i = 1; i < words.length; i++) {
            if (words[i].charAt(0) != words[i - 1].charAt(words[i - 1].length() - 1)) {
                return "false";
            }
        }

        return "true";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Entrada de exemplo
        String sentence = scanner.nextLine();
        
        // Saída de exemplo
        System.out.println(checkSentenceFlow(sentence));  // Saída: true
    }
}

File: Solo Learn/Médio/Palavras Fluentes/flowing_words.rb
def check_sentence_flow(sentence)
  words = sentence.downcase.split  # Converte para minúsculas e divide a frase em palavras

  (1...words.length).each do |i|
    if words[i][0] != words[i - 1][-1]  # Verifica se a primeira letra da palavra atual corresponde à última letra da palavra anterior
      return 'false'
    end
  end

  'true'
end

# Entrada de exemplo
sentence = gets.chomp
# Saída de exemplo
puts check_sentence_flow(sentence)  # Saída: true